core/src/main/java/io/questdb/cairo/VarcharTypeDriver.java

```diff
@@ -173,6 +173,80 @@ public static long getDataVectorSize(MemoryR auxMem, long offset) {
         return dataOffset + size;
     }
 
+    /**
+     * Returns a Utf8Sequence set up to point to the varchar data in either the
+     * auxiliary vector (for short strings <= 9 bytes), or the data vector.
+     *
+     * @param rowNum  the row number to read
+     * @param dataMem base pointer of the data vector
+     * @param auxMem  base pointer of the auxiliary vector
+     * @param ab      whether to return the A or B flyweight
+     * @return a Utf8Sequence representing the value at rowNum
+     */
+    public static DirectUtf8Sequence getDirectValue(long rowNum, MemoryR dataMem, MemoryR auxMem, int ab) {
+        final long auxOffset = VARCHAR_AUX_WIDTH_BYTES * rowNum;
+        int raw = auxMem.getInt(auxOffset);
+        assert raw != 0;
+
+        if (hasNullFlag(raw)) {
+            return null;
+        }
+
+        boolean isAscii = hasAsciiFlag(raw);
+
+        if (hasInlinedFlag(raw)) {
+            int size = (raw >> HEADER_FLAGS_WIDTH) & INLINED_LENGTH_MASK;
+            return ab == 1
+                    ? auxMem.getVarcharA(auxOffset + 1, size, isAscii)
```

Comment:
nit: we should probably use `FULLY_INLINED_STRING_OFFSET` here and on the next line.